### PR TITLE
Develop to Master

### PR DESCRIPTION
**Pull request by amansrivastava17**
- Changes related to `past_date_reference` and year normalization for all languages
- Support for predetected values in text and name detection

### CODE REVIEW

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/detectors/temporal/date/en/date_detection.py [hunk 1]
```diff
@@ -1998,13 +1999,13 @@ def normalize_year(self, year):
...
         this_century = int(str(self.now_date.year)[:2])
         if len(year) == 2:
-            if self.bot_message:
-                if past_regex and past_regex.search(self.bot_message):
-                    return str(this_century - 1) + year
-                elif present_regex and present_regex.search(self.bot_message):
-                    return str(this_century) + year
-                elif future_regex and future_regex.search(self.bot_message):
-                    return str(this_century + 1) + year
+            if (((self.bot_message and past_regex.search(self.bot_message)) or
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: W504 line break after binary operator

File: ner_v2/detectors/temporal/date/standard_date_regex.py [hunk 2]
```diff
@@ -537,6 +538,119 @@ def _detect_weekday(self, date_list, original_list):
...
+            original = pattern[0]
+            dd = int(pattern[1])
+            mm = int(pattern[2])
+            yy = int(self.normalize_year(pattern[3])) if pattern[3] else self.now_date.year
+            try:
+                # to catch dates which are not possible like "31/11" (october 31st)
+                if not pattern[3] and self.timezone.localize(datetime.datetime(year=yy, month=mm, day=dd))\
+                        < self.now_date:
+                    yy += 1
+            except:
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/standard_date_regex.py]: E722 do not use bare 'except'

File: ner_v2/detectors/temporal/date/standard_date_regex.py [hunk 3]
```diff
@@ -537,6 +538,119 @@ def _detect_weekday(self, date_list, original_list):
...
+            str: year in four digits
+        """
+        # past_regex = re.compile(ur'birth|bday|dob|born|जन्म|जन्मदिन|పుట్టినరోజు|పుట్టిన', flags=re.UNICODE)
+        past_regex = None
+        # Todo: Add more language variations of birthday.
+        present_regex = None
+        future_regex = None
+        this_century = int(str(self.now_date.year)[:2])
+        if len(year) == 2:
+            if (((self.bot_message and past_regex and past_regex.search(self.bot_message)) or
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/standard_date_regex.py]: W504 line break after binary operator

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v1/detectors/textual/name/name_detection.py [hunk 4]
```diff
@@ -256,20 +334,20 @@ def replace_detected_text(self, text_detection_result, text):
...
+            replaced_text_tokens = nltk_tokenizer.tokenize(text.lower())
         elif self.language == HINDI_LANG:
-            replaced_text = text.lower().strip().split()
+            replaced_text_tokens = text.lower().strip().split()
 
         for detected_original_text in (text_detection_result[1]):
-            for j in range(len(replaced_text)):
-                replaced_text[j] = replaced_text[j].replace(detected_original_text, "_" + detected_original_text + "_")
+            for j in range(len(replaced_text_tokens)):
+                replaced_text_tokens[j] = replaced_text_tokens[j].replace(detected_original_text, "_" + detected_original_text + "_")
```
Comment (srajdev) [inline ner_v1/detectors/textual/name/name_detection.py]: E501 line too long (133 > 119 characters)

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

File: ner_v1/detectors/textual/name/tests/test_name_detection.py [hunk 5]
```diff
@@ -4,7 +4,10 @@
 
 import pandas as pd
 from django.test import TestCase
+import mock
```
Comment (srajdev) [inline ner_v1/detectors/textual/name/tests/test_name_detection.py]: F401 'mock' imported but unused

File: ner_v1/detectors/textual/name/tests/test_name_detection.py [hunk 6]
```diff
@@ -4,7 +4,10 @@
...
+import json
```
Comment (srajdev) [inline ner_v1/detectors/textual/name/tests/test_name_detection.py]: F401 'json' imported but unused

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by chiragjn** [comment]
retest this please

**Pull request merged**